script.js

```diff
@@ -1,4 +1,435 @@
+  console.log("currentGameMode", currentGameMode);
+  return myOutputValue;
+};
+
+//----------- Set user name game mode --------------------
+
+var setUserName = function (playerName) {
+  userName = playerName;
+  currentGameMode = "Choose your game";
+  var myOutputValue = `Hello ${userName}! Please choose the game mode you'll like to play. <br><br>Type in "Normal", "Reversed", "Korean", "AI"`;
+  return myOutputValue;
+};
+
+// ----------- game mode to choose the type of game---------
+
+var startGame = function (input) {
+  var myOutputValue = `Please type "Normal", "Reversed", "Korean", "AI"`;
+  if (input == "Normal") {
+    currentGameMode = "Normal";
+    myOutputValue = `You have selected ${input} as your game of choice! <br><br>This is the normal scissors, paper, stone game.<br><br>Start by typing in "scissors", "paper" or "stone"`;
+  }
+  if (input == "Reversed") {
+    currentGameMode = "Reversed";
+    myOutputValue = `You have selected ${input} as your game of choice! <br><br>This is the REVERSED of scissors, paper, stone game.<br><br>Start by typing in "scissors", "paper" or "stone"`;
+  }
+  if (input == "Korean") {
+    currentGameMode = "Korean";
+    myOutputValue = `You have selected ${input} as your game of choice! <br><br>The Ultimate Winner is determined in a draw game.<br><br>The Ultimate Winner is the previous winner when the current game is a draw!<br><br>Start by typing in "scissors", "paper" or "stone"`;
+  }
+  if (input == "AI") {
+    currentGameMode = "AI";
+    myOutputValue = `You have selected Computer Vs AI as your game of choice! <br><br>An AI machine will pick for you!<br><br>Start by pressing Submit`;
+  }
+  return myOutputValue;
+};
+
+// ------------------ Normal game mode ----------------------
+
+var normalGame = function (input) {
+  // to generate computer's move. console log to see response and verify outcome.
+  var computerMove = randomItem();
+  console.log("computer", computerMove);
+
+  // to generate the icons for the moves
+  var iconComputer = iconLogo(computerMove);
+  console.log("iconComputer", iconComputer);
+  var iconPlayer = iconLogo(input);
+  console.log("iconPlayer", iconPlayer);
+
+  // to call the win, draw, invalid functions in main to help determine outcome.
+  var draw = drawGame(input, computerMove);
+  var win = playerWin(input, computerMove);
+  var tryAgain = invalid(input);
+
+  //tracking of game progress
+  numGamesPlayed += 1;
+  console.log("numGamesPlayed", numGamesPlayed);
+
+  //set base output as lose scenario
+  var myOutputValue = `The computer played ${computerMove} ${iconComputer} .<br>You played ${input} ${iconPlayer}.<br>You lost ='( <br>${userName}, you have played ${numGamesPlayed} games and your win percentage is ${percentageWin()}%.<br>Lets play another round!`;
+
+  // IF condition to determine if player has input an invalid guess with corresponding output. To minus 1 game to counter the default +1 above
+  if (tryAgain) {
```

Comment:
For what you were trying to do here, it could have been clearer, to just put in
`if (invalid(input)) { `....

Instead of tying that to another variable